libethereum/TransactionReceipt.h

```diff
@@ -52,7 +58,12 @@ class TransactionReceipt
 	bytes rlp() const { RLPStream s; streamRLP(s); return s.out(); }
 
 private:
-	h256 m_stateRoot;
+	bool m_hasStatusCode = false;
+	union
```

Comment:
I'd suggest trying to use `boost::variant` for this, it kind of encapsulates union + type variable